src/components/Project/test/ProjectExecutions.test.tsx

```diff
@@ -20,59 +20,116 @@ import {
+        executions1 = [
             basicPythonFixture.workflowExecutions.top.data,
             failedTaskFixture.workflowExecutions.top.data
         ];
-        const { domain, project } = executions[0].id;
+        executions2 = [];
+        const { domain, project } = executions1[0].id;
         scope = { domain, project };
         mockServer.insertWorkflowExecutionList(
             scope,
-            executions,
-            defaultQueryParams
+            executions1,
+            defaultQueryParams1
+        );
+        mockServer.insertWorkflowExecutionList(
+            scope,
+            executions2,
+            defaultQueryParams2
         );
     });
 
     const renderView = () =>
         render(
             <QueryClientProvider client={queryClient}>
-                <ProjectExecutions
-                    projectId={scope.project}
-                    domainId={scope.domain}
-                />
+                <APIContext.Provider
+                    value={mockAPIContextValue({
+                        getUserProfile: mockGetUserProfile
+                    })}
+                >
+                    <ProjectExecutions
+                        projectId={scope.project}
+                        domainId={scope.domain}
+                    />
+                </APIContext.Provider>
             </QueryClientProvider>,
             { wrapper: MemoryRouter }
         );
 
-    it('displays executions after successful load', async () => {
-        const { getByText } = renderView();
-        await waitFor(() => expect(getByText(executions[0].id.name)));
+    it('should not display checkbox if user does not login', async () => {
+        const { queryByTestId } = renderView();
+        await waitFor(() => {});
+        expect(mockGetUserProfile).toHaveBeenCalled();
+        expect(queryByTestId(/checkbox/i)).toBeNull();
+    });
+
+    it('should display checkbox if user login', async () => {
+        mockGetUserProfile.mockResolvedValue(sampleUserProfile);
+        const { queryByTestId } = renderView();
+        await waitFor(() => {});
+        expect(mockGetUserProfile).toHaveBeenCalled();
+        const checkbox = queryByTestId(/checkbox/i) as HTMLInputElement;
+        expect(checkbox).toBeTruthy();
     });
 
-    it('shows name of corresponding workflow in row items', async () => {
-        const { getByText } = renderView();
+    /** user doesn't have its own workflow */
+    it('should not display workflow if user does not have workflow', async () => {
+        mockGetUserProfile.mockResolvedValue(sampleUserProfile);
+        const { getByText, queryByText, queryByTestId } = renderView();
+        await waitFor(() => {});
+        expect(mockGetUserProfile).toHaveBeenCalled();
+        const checkbox = queryByTestId(/checkbox/i)?.querySelector(
+            'input'
+        ) as HTMLInputElement;
+        expect(checkbox).toBeTruthy();
+        expect(checkbox?.checked).toEqual(true);
+        await waitFor(() =>
```

Comment:
Also recommending here: 
* Put the assertion on line 118 into the `waitFor` above it.
* Use `getByLabelText` to get the checkbox
* Use `toBeChecked` to check if the checkbox is checked